cmd/kata-manager/kata-manager.sh

```diff
@@ -0,0 +1,328 @@
+
+typeset -r doc_repo="github.com/kata-containers/documentation"
+typeset -r test_repo="github.com/kata-containers/tests"
+
+typeset -r config_file_name="configuration.toml"
+typeset -r default_image_file="/usr/share/kata-containers/kata-containers.img"
+typeset -r default_config_file="/usr/share/defaults/kata-containers/${config_file_name}"
+typeset -r local_config_file="/etc/kata-containers/${config_file_name}"
+
+# kernel boot option to enable agent debug
+typeset -r agent_debug="agent.log=debug"
+
+verbose="no"
+
+# full path to the runtime configuration file to operate on
+config_file=
+
+# lower-case name of distribution
+distro=
+
+usage()
+{
+	cat <<EOT
+Usage: ${script_name} [options] [command]
+
+Description: Install and configure Kata Containers.
+
+Options:
+
+  -c <file> : Specify full path to configuration file
+              (default: '$local_config_file').
+  -h        : Display this help.
+  -v        : Verbose output.
+
+Commands:
+
+  configure-image  : Configure the runtime to use the specified image.
+  configure-initrd : Configure the runtime to use the specified initial ramdisk.
+  disable-debug    : Turn off all debug options.
+  enable-debug     : Turn on all debug options for all system components.
+  install-packages : Install the packaged version of Kata Containers.
+  remove-packages  : Uninstall the packaged version of Kata Containers.
+  reset-config     : Undo changes to the runtime configuration [1].
+
+Notes:
+
+[1] - This is only possible if the user has not specified '-c'.
+
+
+EOT
+}
+
+die()
+{
+	local msg="$*"
+	echo "ERROR: $msg" >&2
+	exit 1
+}
+
+info()
+{
+	[ "$verbose" != "yes" ] && return
+
+	local msg="$*"
+	echo "INFO: $msg"
+}
+
+check_file()
+{
+	local file="$1"
+
+	[ -z "$file" ] && die "need file"
+
+	# Need to check as root to avoid EPERM which will cause the test to
+	# fail
+	sudo test ! -e "$file" && die "file '$file' does not exist" || true
+}
+
+config_checks()
+{
+	[ -e "$local_config_file" ] && return
+
+	# The user has specified a non-standard config file so the only way to
+	# handle this is to edit that file itself.
+	[ "$config_file" != "$local_config_file" ] && return
+
+	local dir=$(dirname "$local_config_file")
+
+	sudo mkdir -p "$dir"
+
+	# Create a copy of the default config file that this script can
+	# operate on, but only if that file doesn't already exist.
+	#
+	# Note: these permissions match the currently packaged permissions
+	if [ ! -e "$local_config_file" ]
+	then
+		sudo install -o root -g root -m 0644 \
+			"$default_config_file" \
+			"$local_config_file"
+
+		# use an image by default
+		cmd_configure_image "$default_image_file"
+	fi
+}
+
+disable_image()
+{
+	config_checks
+
+	sudo sed -i 's/^\(image *=.*\)/# \1/g' "$config_file"
+}
+
+disable_initrd()
+{
+	config_checks
+
+	sudo sed -i 's/^\(initrd *=.*\)/# \1/g' "$config_file"
+}
+
+enable_image()
+{
+	local file="$1"
+
+	config_checks
+
+	sudo sed -i "s!^#*.*image *=.*\$!image = \"$file\"!g" "$config_file"
```

Comment:
The image will be modified if is commented , should the regex also replace if the image is not commented?

Reply:
It actually will since the first part of the regex is `#*` - aka "zero or more hashes", so the hash can be absent.